### PR TITLE
Feature/addition of products categories list

### PR DESCRIPTION
**Pull request by pasternakewa**

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -10,6 +10,7 @@ body {
   height: 100vh;
   padding: 0px;
   font-family: "Montserrat", sans-serif;
+  overflow-x: hidden;
```
Comment (vdanylov) [inline src/App.css:12]: why only x??

File: src/App.css [hunk 2]
```diff
@@ -56,52 +57,117 @@ nav {
...
-  object-fit: cover;
+  z-index: -2;
 }
 
-.gallery__element--1 {
-  grid-column: 1 / 2;
-  grid-row: 1 / 2;
+.gallery-item {
+  float: right;
+  z-index: 5;
```
Comment (vdanylov) [inline src/App.css]: is layout could be handled without z-index in this case?

File: src/App.css [hunk 3]
```diff
@@ -56,52 +57,117 @@ nav {
...
+  position: relative;
+  top: 1px;
 }
 
-.gallery__element--4 {
-  grid-column: 1 / 2;
-  grid-row: 2/ 3;
+.gallery-item__img {
+  box-shadow: rgba(0, 0, 0, 0.15) 0px 5px 15px 0px;
+  z-index: -1;
```
Comment (vdanylov) [inline src/App.css]: is layout could be handled without z-index in this case?

File: src/App.css [hunk 4]
```diff
@@ -56,52 +57,117 @@ nav {
...
+  position: relative;
+  width: 100%;
+  max-width: 100%;
```
Comment (vdanylov) [inline src/App.css]: do we need here max-width?

File: src/components/Products.js [hunk 5]
```diff
@@ -1,9 +1,51 @@
...
+import Image from "./Image";
 
 export default function Products() {
+  const [items, setItems] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    fetchItems();
+    return () => {
+      setItems([]);
```
Comment (vdanylov) [inline src/components/Products.js]: what is the purpose of this callback?

File: src/components/Products.js [hunk 6]
```diff
@@ -1,9 +1,51 @@
...
+      console.log("error:", e);
+      setIsLoading(false);
+    }
+  };
+
+  let uniqueItems = [
+    ...new Map(items.map((item) => [item.product_type, item])).values()
+  ];
+
+  console.log("uniqueItems", uniqueItems);
```
Comment (vdanylov) [inline src/components/Products.js]: remove

File: src/components/Products.js [hunk 7]
```diff
@@ -1,9 +1,51 @@
...
+      setItems(items);
+      setIsLoading(false);
+    } catch (e) {
+      console.log("error:", e);
+      setIsLoading(false);
+    }
+  };
+
+  let uniqueItems = [
+    ...new Map(items.map((item) => [item.product_type, item])).values()
```
Comment (vdanylov) [inline src/components/Products.js:28]: what it does?

File: src/App.css (hunk 1 above)
Comment (pasternakewa) [reply src/App.css:12]: because scrolling up and down must be enabled

File: src/App.css (hunk 3 above)
Comment (pasternakewa) [reply src/App.css]: Unfortunately it could not. The caption is on the same level as img.

File: src/App.css (hunk 2 above)
Comment (pasternakewa) [reply src/App.css]: Unfortunately without this z-index the img transition does not work properly :-(

File: src/components/Products.js (hunk 7 above)
Comment (pasternakewa) [reply src/components/Products.js:28]: Would getUniqueCategoryProducts() be better name?

File: src/components/Products.js (hunk 5 above)
Comment (pasternakewa) [reply src/components/Products.js]: no purpose :) done :)

**Pull request merged**